streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorTopologyTest.java

```diff
@@ -134,77 +136,97 @@ public void testTopologyMetadata() {
     public void testDrivingSimpleTopology() {
         final int partition = 10;
         driver = new TopologyTestDriver(createSimpleTopology(partition), props);
-        driver.pipeInput(recordFactory.create(INPUT_TOPIC_1, "key1", "value1"));
-        assertNextOutputRecord(OUTPUT_TOPIC_1, "key1", "value1", partition);
-        assertNoOutputRecord(OUTPUT_TOPIC_2);
+        final TestInputTopic<String, String> inputTopic = driver.createInputTopic(INPUT_TOPIC_1, STRING_SERIALIZER, STRING_SERIALIZER, Instant.ofEpochMilli(0L), Duration.ZERO);
+        final TestOutputTopic<String, String> outputTopic1 =
+                driver.createOutputTopic(OUTPUT_TOPIC_1, Serdes.String().deserializer(), Serdes.String().deserializer());
+        final TestOutputTopic<String, String> outputTopic2 =
+                driver.createOutputTopic(OUTPUT_TOPIC_1, Serdes.String().deserializer(), Serdes.String().deserializer());
 
-        driver.pipeInput(recordFactory.create(INPUT_TOPIC_1, "key2", "value2"));
-        assertNextOutputRecord(OUTPUT_TOPIC_1, "key2", "value2", partition);
-        assertNoOutputRecord(OUTPUT_TOPIC_2);
+        inputTopic.pipeInput("key1", "value1");
+        assertNextOutputRecord(outputTopic1.readRecord(), "key1", "value1");
+        assertTrue(outputTopic1.isEmpty());
 
-        driver.pipeInput(recordFactory.create(INPUT_TOPIC_1, "key3", "value3"));
-        driver.pipeInput(recordFactory.create(INPUT_TOPIC_1, "key4", "value4"));
-        driver.pipeInput(recordFactory.create(INPUT_TOPIC_1, "key5", "value5"));
-        assertNoOutputRecord(OUTPUT_TOPIC_2);
-        assertNextOutputRecord(OUTPUT_TOPIC_1, "key3", "value3", partition);
-        assertNextOutputRecord(OUTPUT_TOPIC_1, "key4", "value4", partition);
-        assertNextOutputRecord(OUTPUT_TOPIC_1, "key5", "value5", partition);
```

Comment:
Why did we eliminate the partition check in the `assertNextOutputRecord` method?

Reply:
Partitions are not part of the `TestRecord` API in the KIP, so it's not possible to make assertions on it in the new API. We can add the field later on if requested.

I don't remember offhand why we didn't include it; maybe something to do with the symmetry of the API.

Reply:
I removed all unnecessary field from TestRecord compared to ProducerRecord and ConsumerRecord to keep it simple. This way we can utilize standard assertions and do not need to ignore the partition and use OutputVerifier kind of contructions.

I don't see many use cases in normal Stream Application verification where partion is needed as we see here where this is only test class checking partition.

If there are need to verify partition, I would add for example extra method readProducerRecord and use deprecated method until this is possible added if there is need for it.